File: Examenes/examen1/QytetetJava/src/modeloqytetet/Casilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author mati
 */
public class Casilla {
    
    private int numeroCasilla;
    private int coste;
    private int numHoteles;
    private int numCasas;
    private TipoCasilla tipo; //los de tipo calle tienen hoteles y casas, el resto no?
    private TituloPropiedad titulo;

    public Casilla(int numeroCasilla, int coste, int numHoteles, int numCasas, TipoCasilla tipo, TituloPropiedad titulo) {
        this.numeroCasilla = numeroCasilla;
        this.coste = coste;
        this.numHoteles = numHoteles;
        this.numCasas = numCasas;
        this.tipo = tipo;
        this.titulo = titulo;
    }

    public Casilla(int numeroCasilla, int coste, TipoCasilla tipo) {
        this.numeroCasilla = numeroCasilla;
        this.coste = coste;
        this.tipo = tipo;
        this.titulo = null;
        this.numHoteles = 0;
        this.numCasas = 0;
    }

    public TipoCasilla getTipo() {
        return this.tipo;
    }
    
    public TituloPropiedad getTituloPropiedad() {
        return titulo;
    }
    
    /*
    TituloPropiedad asignarPropietario( Jugador jugador){
    
    }
    
    int calcularValorHipoteca(){
    
    }
    
    int cancelarHipoteca(){
    
    }
    
    int cobrarAlquiler(){
    
    }
    
    int edificarCasa(){
    
    }
    
    int edificarHotel(){
    
    }
    
    boolean estaHipotecada(){
    
    } 
    */
    
    int getCoste() {
        return coste;
    }
    
    /*
    int getCosteHipoteca(){
    
    }
    */
    
    int getNumeroCasilla() {
        return numeroCasilla;
    }

    int getNumHoteles() {
        return numHoteles;
    }

    int getNumCasas() {
        return numCasas;
    }
    
    /*
    int getPrecioEdificar({
    
    }
    
    int hipotecar(){
    
    }
    
    int precioTotalComprar(){
    
    }
    
    boolean propietarioEncarcelado(){
    
    }
    
    boolean sePuedeEdificarCasa(){
    
    }
    
    boolean sePuedeEdificarHotel(){
    
    }
    */

    void setNumHoteles(int numHoteles) {
        this.numHoteles = numHoteles;
    }

    void setNumCasas(int numCasas) {
        this.numCasas = numCasas;
    }
    
    /*
    boolean soyEdificable(){
    
    }
    
    boolean tengoPropietario(){
    
    }
    
    int venderTitulo(){
    
    }
    
    */

    private void setTituloPropiedad(TituloPropiedad titulo) {
        this.titulo = titulo;
    }
    
    /*
    private void asignarTituloPropiedad(){
    
    }
    */

    @Override
    public String toString() {
        return "Casilla{" + "numeroCasilla=" + numeroCasilla + ",\n coste=" + coste +
                ",\n numHoteles=" + numHoteles + ",\n numCasas=" + numCasas +
                ",\n tipo Casilla=" + tipo + ",\n titulo en propiedad=" + titulo + '}';
    }
    
    
    
}

File: Examenes/examen1/QytetetRuby/lib/casilla.rb
 #encoding: utf-8
 # To change this license header, choose License Headers in Project Properties.
 # To change this template file, choose Tools | Templates
 # and open the template in the editor.
 
 require_relative "tipo_casilla"
 require_relative "titulo_propiedad"

module ModeloQytetet
  class Casilla
    
    attr_accessor :numeroCasilla, :coste, :numHoteles, :numCasas, :tipo, :titulo
       
    def initialize(numCasilla,cos,numH,numCasa, tip, titul)  # dos constructores
      @numeroCasilla = numCasilla  # int
      @coste = cos  # int, coste de compra del titulo de propiedad de la casilla
      @numHoteles = numH  # int
      @numCasas = numCasa  # int
      @tipo = tip  # TipoCasilla
      @titulo = titul  # TituloPropiedad
      if @titulo != nil 
        @titulo.casilla = self 
      end
      
    end
    
    def self.crear_casillas_calle(numCasilla,cos, titul)
      new(numCasilla,cos, 0, 0, TipoCasilla::CALLE, titul)
    end
    
    def self.crear_casillas_especiales(numCasilla, tip)
      new(numCasilla,0, 0, 0, tip, nil)
    end
    
    def asignarPropietario(jugador) #TituloPropiedad asignarPropietario( Jugador jugador)
      raise ArgumentError,'metodo sin implementar'
    end
    
    def calcularValorHipoteca #int calcularValorHipoteca()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def cancelarHipoteca #int cancelarHipoteca()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def cobrarAlquiler #int cobrarAlquiler()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def edificarCasa #int edificarCasa()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def edificarHotel #int edificarHotel()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def estaHipotecada #boolean estaHipotecada()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def getPrecioEdificar #int getPrecioEdificar()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def hipotecar #int hipotecar()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def precioTotalComprar #int precioTotalComprar()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def propietarioEncarcelado #boolean propietarioEncarcelado()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def sePuedeEdificarCasa #boolean sePuedeEdificarCasa()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def sePuedeEdificarHotel #boolean sePuedeEdificarHotel()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def soyEdificable #boolean soyEdificable()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def tengoPropietario #boolean tengoPropietario()
      raise ArgumentError,'metodo sin implementar'
    end
    
    def venderTitulo #int venderTitulo()
      raise ArgumentError,'metodo sin implementar'
    end
    
    private
    
    def asignarTituloPropiedad #private void asignarTituloPropiedad()
      raise ArgumentError,'metodo sin implementar'
    end
    
    public
    
    def to_s
      "numeroCasilla: #{@numeroCasilla} \n 
      coste: #{@coste} \n 
      numHoteles: #{@numHoteles} \n
      numCasas: #{@numCasas} \n 
      tipo: #{@tipo} \n "
      if(@titulo != nil)
        "titulo: #{@titulo} "
      else
        "titulo: no tiene titulo"
      end
    end
    
    private:titulo= ###########################################################################3
    
    
  end
end
